x/sync/network_server_test.go

```diff
@@ -27,116 +28,115 @@ func Test_Server_GetRangeProof(t *testing.T) {
 	t.Logf("seed: %d", now)
 	r := rand.New(rand.NewSource(now)) // #nosec G404
 
-	smallTrieDB, _, err := generateTrieWithMinKeyLen(t, r, defaultRequestKeyLimit, 1)
+	smallTrieDB, err := generateTrieWithMinKeyLen(t, r, defaultRequestKeyLimit, 1)
 	require.NoError(t, err)
 	smallTrieRoot, err := smallTrieDB.GetMerkleRoot(context.Background())
 	require.NoError(t, err)
 
-	tests := map[string]struct {
+	tests := []struct {
+		name                     string
 		request                  *pb.SyncGetRangeProofRequest
-		expectedErr              error
+		expectedErr              *common.AppError
 		expectedResponseLen      int
 		expectedMaxResponseBytes int
 		nodeID                   ids.NodeID
 		proofNil                 bool
 	}{
-		"proof too large": {
+		{
+			name: "proof too large",
 			request: &pb.SyncGetRangeProofRequest{
 				RootHash:   smallTrieRoot[:],
 				KeyLimit:   defaultRequestKeyLimit,
 				BytesLimit: 1000,
 			},
 			proofNil:    true,
-			expectedErr: ErrMinProofSizeIsTooLarge,
+			expectedErr: p2p.ErrUnexpected,
 		},
-		"byteslimit is 0": {
+		{
+			name: "byteslimit is 0",
 			request: &pb.SyncGetRangeProofRequest{
 				RootHash:   smallTrieRoot[:],
 				KeyLimit:   defaultRequestKeyLimit,
 				BytesLimit: 0,
 			},
-			proofNil: true,
+			proofNil:    true,
+			expectedErr: p2p.ErrUnexpected,
 		},
-		"keylimit is 0": {
+		{
+			name: "keylimit is 0",
 			request: &pb.SyncGetRangeProofRequest{
 				RootHash:   smallTrieRoot[:],
-				KeyLimit:   defaultRequestKeyLimit,
-				BytesLimit: 0,
+				KeyLimit:   0,
+				BytesLimit: defaultRequestByteSizeLimit,
 			},
-			proofNil: true,
+			proofNil:    true,
+			expectedErr: p2p.ErrUnexpected,
 		},
-		"keys out of order": {
+		{
+			name: "keys out of order",
 			request: &pb.SyncGetRangeProofRequest{
 				RootHash:   smallTrieRoot[:],
 				KeyLimit:   defaultRequestKeyLimit,
 				BytesLimit: defaultRequestByteSizeLimit,
 				StartKey:   &pb.MaybeBytes{Value: []byte{1}},
 				EndKey:     &pb.MaybeBytes{Value: []byte{0}},
 			},
-			proofNil: true,
+			proofNil:    true,
+			expectedErr: p2p.ErrUnexpected,
 		},
-		"key limit too large": {
+		{
+			name: "response bounded by key limit",
 			request: &pb.SyncGetRangeProofRequest{
 				RootHash:   smallTrieRoot[:],
 				KeyLimit:   2 * defaultRequestKeyLimit,
 				BytesLimit: defaultRequestByteSizeLimit,
 			},
 			expectedResponseLen: defaultRequestKeyLimit,
 		},
-		"bytes limit too large": {
+		{
+			name: "response bounded by byte limit",
```

Comment:
it's the same test but we renamed it because the previous name makes it sound like it's supposed to error when the expected behavior is to truncate/cap the response from the server.

Reply:
We could have a case where the bytes limit requested by the client is valid (less than the limit enforced by the server) and is the limiting factor in the response size.